drivers: sensor: Add suport for LTR329 sensor

konrad1s opened the pull request:
Hello
this pull request adds support the Liteon LTR329 Digital Ambient Light Sensor with I2C bus interface.

Datasheet for the sensor is available at the link: [datasheet](https://www.mouser.com/datasheet/2/239/Lite-On_LTR-329ALS-01%20DS_ver1.1-348647.pdf)
The ambient light conversion equation is as shown in the following document: [LTR-303ALS & LTR-329ALS Appendix A](https://cdck-file-uploads-europe1.s3.dualstack.eu-west-1.amazonaws.com/arduino/original/3X/8/d/8ddfc2d335f178e70e20974439a94ce31dd9df35.pdf) 
I have adapted the ambient light conversion equation to use fixed-point calculations instead of floating-point, enhancing efficiency.

konrad1s committed:
drivers: sensor: Add suport for LTR329 sensor

Introduce support for Liteon LTR329 Digital Ambient Light Sensor

Signed-off-by: Konrad Sikora <konradsikora98@gmail.com>

kartben commented on drivers/sensor/ltr329/ltr329.c:
```diff
@@ -0,0 +1,359 @@
+static int ltr329_get_mapped_int_time(const uint8_t register_val, uint8_t *const output)
+{
+	/* Map the register value to the integration time used in the lux calculation */
+	const int32_t int_time_lux_calc[] = {10, 5, 20, 40, 15, 25, 30, 35};
```
```suggestion
	const uint8_t int_time_lux_calc[] = {10, 5, 20, 40, 15, 25, 30, 35};
```

MaureenHelm commented on dts/bindings/sensor/ltr,329.yaml:
```diff
@@ -0,0 +1,59 @@
+properties:
+  gain:
+    type: int
+    default: 0
```
Need to justify in the description why this default value was chosen. Is it a register reset value?

konrad1s replied:
register reset value, comment added

MaureenHelm commented on dts/bindings/sensor/ltr,329.yaml:
```diff
@@ -0,0 +1,59 @@
+
+  integration-time:
+    type: int
+    default: 0
```
Need to justify in the description why this default value was chosen. Is it a register reset value?

konrad1s replied:
register reset value, comment added

MaureenHelm commented on dts/bindings/sensor/ltr,329.yaml:
```diff
@@ -0,0 +1,59 @@
+
+  measurement-rate:
+    type: int
+    default: 3
```
Need to justify in the description why this default value was chosen. Is it a register reset value?

konrad1s replied:
register reset value, comment added

konrad1s committed:
drivers: sensor: Change of vendor prefix from 'ltr' to 'liteon'

The vendor prefix has been changed from 'ltr' to 'liteon'. This change is made to better reflect the actual name of the vendor, which is LiteOn.
The previous prefix 'ltr' was specifically referring to the LiteOn Optical Sensor family, which could lead to misunderstandings.

Signed-off-by: Konrad Sikora <konradsikora98@gmail.com>

konrad1s committed:
drivers: sensor: Review changes - to be squashed

Signed-off-by: Konrad Sikora <konradsikora98@gmail.com>

konrad1s commented:
Hello, 
I fixed all review comments (changes added to the new commit).
I also changed the vendor from 'ltr' to 'liteon'. The previous prefix 'ltr' was specifically referring to the LiteOn Optical Sensor family [link](https://optoelectronics.liteon.com/en-global/led/LED-Component) which could lead to misunderstandings.

I intended to work on implementing an emulator, but I'm still grappling with some problems. The main one is lack of time. I will try to do this (if its not required, perhaps in the next PR)

konrad1s commented on drivers/sensor/ltr329/ltr329.c:
```diff
@@ -0,0 +1,358 @@
+			LOG_ERR("Failed to get LTR329 ALS_STATUS");
+			return rc;
+		}
+		k_sleep(K_MSEC(3));
```
removed, return -EBUSY added (I saw a similar solution in other sensors)

konrad1s commented on dts/bindings/sensor/ltr,329.yaml:
```diff
@@ -0,0 +1,59 @@
+
+description: Lite-on LTR329 Digital Ambient Light Sensor
+
+compatible: "ltr,329"
```
Updated, changed vendor from ltr to liteon
